manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/dao/ConnectorDAO.java

```diff
@@ -35,6 +45,33 @@ public Connector findByBridgeIdAndName(String bridgeId, String name) {
         return find("#CONNECTOR_V2.findByBridgeIdAndName", params).firstResult();
     }
 
+    public ListResult<Connector> findByBridgeIdAndCustomerId(String bridgeId, String customerId, QueryResourceInfo queryInfo) {
+        Parameters parameters = Parameters.with(Processor.BRIDGE_ID_PARAM, bridgeId).and(Bridge.CUSTOMER_ID_PARAM, customerId);
+        PanacheQuery<Connector> query = find("#CONNECTOR_V2.findByBridgeIdAndCustomerId", parameters);
+
+        String filterName = queryInfo.getFilterInfo().getFilterName();
+        if (Objects.nonNull(filterName)) {
+            query.filter("byName", Parameters.with(Connector.NAME_PARAM, filterName + "%"));
+        }
+
+        // As the status of the resource is a view over the conditions, it has to be calculated on the fly. ATM we do it in java, in case there are
+        // performance issues we might move the filtering to the database adapting the database schema accordingly.
+
+        List<Connector> filtered = query.list();
+        Set<ManagedResourceStatusV2> filterStatus = queryInfo.getFilterInfo().getFilterStatus();
+        if (Objects.nonNull(filterStatus) && !filterStatus.isEmpty()) {
+            // Calculate the status of the resource and apply filter.
+            filtered = filtered.stream().filter(x -> filterStatus.contains(StatusUtilities.getManagedResourceStatus(x))).collect(Collectors.toList());
+        }
+
+        long total = filtered.size();
+        int startIndex = queryInfo.getPageNumber() * queryInfo.getPageSize();
+        int endIndex = startIndex + queryInfo.getPageSize();
+
+        List<Connector> processors = startIndex >= total ? new ArrayList<>() : filtered.subList(startIndex, (int) Math.min(total, endIndex));
```

Comment:
I wonder where you copied and pasted this code from!?!?

Perhaps you want to change the variable `processors` to `connectors`?